src/platform_impl/web/event_loop/runner.rs

```diff
@@ -183,11 +183,24 @@ impl<T: 'static> Shared<T> {
         }
         self.init();
 
+        let show_runner = self.clone();
+        let hide_runner = self.clone();
         let close_instance = self.clone();
-        *self.0.unload_event_handle.borrow_mut() =
-            Some(backend::on_unload(self.window(), move || {
-                close_instance.handle_unload()
-            }));
+        *self.0.page_transition_event_handle.borrow_mut() = Some(backend::on_page_transition(
+            self.window(),
+            move |event: PageTransitionEvent| {
+                if event.persisted() {
+                    hide_runner.send_event(Event::Resumed);
+                }
```

Comment:
This should never happen unless it was persisted anyway.
We should turn this into a `debug_assert!` even.

Reply:
A `pageshow` event with `persisted` being `true` is received on "normal" page loads - that is, when loading the page directly, not from the bfcache.

I made a small logger of `pageshow` and `pagehide` events here: https://fornwall.net/bfcache

Loading the page directly, then leaving it (putting it in the bfache), and navigating back (restoring it from the bfcache) shows the three events:

```js
// (1) Normal page load:
{"persisted":false,"type":"pageshow"}

// (2) Leaving the page, putting it in the bfache:
{"persisted":true,"type":"pagehide"}

// (3) Going back to the page, restoring it from the bfcache:
{"persisted":true,"type":"pageshow"}
```

Reply:
> It is true that it could fire as part of the initial page load, but I couldn't get it to fire from Winit (when I removed the if check). By any chance would you know why?

@daxpedda A, good point! I guess this comes from the Winit wasm module being loaded async, so once we setup the event listener the initial `pageshow` has already fired?

But as you say, we should probably leave this code as it is now - it's more of an implementation detail that we are not catching the initial `pageshow` when loaded async, and in the future more wasm code might be loaded on the main thread (see https://chromestatus.com/feature/5099433642950656), so winit might actually receive the initial non-persisted `pageshow` event.

Reply:
Thank you for digging into this.